qemu/tests/timerdevice_boot.py

```diff
@@ -85,96 +66,143 @@ def verify_guest_clock_source(session, expected):
-        return
-
-    # Command to run to get the current time
-    time_command = params["time_command"]
-    # Filter which should match a string to be passed to time.strptime()
-    time_filter_re = params["time_filter_re"]
-    # Time format for time.strptime()
-    time_format = params["time_format"]
-    timerdevice_drift_threshold = params.get("timerdevice_drift_threshold", 3)
-
-    error.context("Check the system time on guest and host", logging.info)
-    (host_time, guest_time) = utils_test.get_time(session, time_command,
-                                                  time_filter_re, time_format)
-    drift = abs(float(host_time) - float(guest_time))
-    if drift > timerdevice_drift_threshold:
-        raise error.TestFail("The guest's system time is different with"
-                             " host's. Host time: '%s', guest time:"
-                             " '%s'" % (host_time, guest_time))
-
-    get_hw_time_cmd = params.get("get_hw_time_cmd")
-    if get_hw_time_cmd:
-        error.context(
-            "Check the hardware time on guest and host", logging.info)
-        host_time = utils.system_output(get_hw_time_cmd)
-        guest_time = session.cmd(get_hw_time_cmd)
-        drift = abs(float(host_time) - float(guest_time))
-        if drift > timerdevice_drift_threshold:
-            raise error.TestFail("The guest's hardware time is different with"
-                                 " host's. Host time: '%s', guest time:"
-                                 " '%s'" % (host_time, guest_time))
-
-    if params.get("timerdevice_reboot_test") == "yes":
-        sleep_time = params.get("timerdevice_sleep_time")
-        if sleep_time:
-            error.context("Sleep '%s' secs before reboot" % sleep_time,
-                          logging.info)
-            sleep_time = int(sleep_time)
-            time.sleep(sleep_time)
+    utils.system("ntpdate clock.redhat.com")
+
+    timerdevice_host_load_cmd = params.get("timerdevice_host_load_cmd")
+    if timerdevice_host_load_cmd:
+        error_context.context("Add some load on host", logging.info)
+        utils.system(timerdevice_host_load_cmd)
+
+    error_context.context("Boot a guest with kvm-clock", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=timeout)
 
-        session = vm.reboot()
-        error.context("Check the system time on guest and host", logging.info)
+    timerdevice_clksource = params.get("timerdevice_clksource")
+    try:
+        if timerdevice_clksource:
+            try:
+                verify_guest_clock_source(session, timerdevice_clksource)
+            except Exception:
+                clksrc = timerdevice_clksource
+                error_context.context("Shutdown guest")
+                vm.destroy()
+                env.unregister_vm(vm.name)
+                update_clksrc_with_kernel_line(clksrc)
```

Comment:
Better include it within the commit where the function defined, since you removed the whole process there.